src/coreclr/jit/gentree.cpp

```diff
@@ -16115,6 +16115,11 @@ bool Compiler::gtTreeHasSideEffects(GenTree* tree, GenTreeFlags flags /* = GTF_S
         return false;
     }
 
+    if (tree->OperIs(GT_RET_EXPR))
+    {
+        return gtTreeHasSideEffects(tree->AsRetExpr()->gtInlineCandidate, flags);
+    }
```

Comment:
Is this necessary for something? We give `GT_RET_EXPR` `GTF_CALL` already so I wouldn't think this would have any effect.

Reply:
@jakobbotsch you're right, just checked